assembly/massa.ts

```diff
@@ -188,12 +202,26 @@ export function get_data_or_default(key: string, default_value: string): string
  * @param default_value default value if not found
  * @returns found string value or default string
  */
-export function get_data_or_default_for(address:string, key: string, default_value: string): string {
-    if(has_data_for(address, key)) {
-        return get_data_for(address, key);
+export function raw_get_data_or_default_for(address: string, key: string, default_value: string): string {
+    if (has_data_for(address, key)) {
+        return raw_get_data_for(address, key);
     }
     return default_value;
 }
+
+export function get_data_or_default_for(address: string, key: string, default_value: string): JSON.Obj {
+    return <JSON.Obj>(JSON.parse(get_data_or_default_for(address, key, default_value)));
+}
+
+export function get_msg_sender(): string {
+    const addresses = get_call_stack();
+    return addresses[addresses.length - 1].toString();
```

Comment:
```suggestion
    return addresses[addresses.length - 2].toString();
```
I think -1 is the current smart-contract, we should take -2 to get the address that called the smart-contract.